include/pressio/ops/epetra/ops_dot.hpp

```diff
@@ -51,26 +51,50 @@
 
 namespace pressio{ namespace ops{
 
-template <typename vec_type>
+template <typename T1, typename T2, typename DotResult>
 ::pressio::mpl::enable_if_t<
-  ::pressio::is_vector_epetra<vec_type>::value
+  // dot common constraints
+     ::pressio::Traits<T1>::rank == 1
+  && ::pressio::Traits<T2>::rank == 1
```

Comment:
no need the check on rank

Reply:
ok, fixed on e7e6bdd233734fad324d7cfb2a97627a8fcf55a4